cirq-core/cirq/transformers/gauge_compiling/iswap_gauge.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A Gauge transformer for ISWAP gate."""
+
+import numpy as np
+
+from cirq.transformers.gauge_compiling.gauge_compiling import (
+    ConstantGauge,
+    Gauge,
+    GaugeTransformer,
+    GaugeSelector,
+)
+from cirq import ops
+
+
+class RZRotation(Gauge):
+    """Represents an ISWAP Gauge composed of Rz rotations.
+
+    The gauge replaces an ISWAP gate with either
+        0: ───Rz(t)──────iSwap───Rz(sgn*t)───
+                            │
+        1: ───Rz(sgn*t)───iSwap───Rz(t)───
+
+    where t is uniformly sampled from [0, 2π) and sgn is uniformly sampled from {-1, 1}.
+    """
```

Comment:
Are you sure this is correct? It differs from what is in the note.

Reply:
fixed